### DISCUSSION TITLE
Comparison with aerc

### DISCUSSION THREAD

**Discussion by alexherbo2** [pimalaya/himalaya#56] [Show and tell]
https://aerc-mail.org

**Comment by soywod** [comment]
Well, it seems to be a TUI (more like mutt/neomutt). Himalaya is a CLI, so they can't really be compared.

**Reply by alexherbo2** [reply]
Yep, I realized that after reading a bit more the project.

It looks pretty solid and composable way of doing things.

**Reply by srid** [reply]
Yea, I really like himalaya's [SoC](https://en.wikipedia.org/wiki/Separation_of_concerns) - where the email "engine" is decoupled from the "view". This allows us to swap out a view for another one. Like, going from TUI to a script interface or editor plugin or even a GUI app (or a web app if using himalaya as a library). The "engine" only needs to provide some API interface, which is typically `--json` output. [Neuron](https://neuron.zettel.page/editor) works on the similar principle.
